Utilities/JIT.cpp

```diff
@@ -1350,6 +1353,18 @@ std::string jit_compiler::cpu(const std::string& _cpu)
 			// Upgrade
 			m_cpu = "znver2";
 		}
+
+		if (m_cpu == "znver3" && utils::has_avx512_icl())
+		{
+			// Upgrade (presumably znver4)
+			m_cpu = "icelake-client";
+		}
+
+		if (m_cpu == "goldmont" && utils::has_avx2())
+		{
+			// Upgrade (presumably raptorlake)
+			m_cpu = "alderlake";
+		}
```

Comment:
LLVM 16 should now properly detect these archs, but we likely need a hack to "upgrade" alderlake with avx-512 to icelake-client added.

Also: the upcoming release of 16.0.1 has the znver4 scheduling model merged.